editor/src/clj/editor/tile_map.clj

```diff
@@ -135,6 +135,153 @@
+                                        (reverse (subvec flipped-tiles (* y width) (* (inc y) width))))))]
+    {:width width
+     :height height
+     :tiles reordered-tiles}))
+
+(defn flip-brush-vertically
+  [brush]
+  (let [width (:width brush)
+        height (:height brush)
+        tiles (:tiles brush)
+        ;; Vertical Flip Mappings
+        ;; | current-state      | hflip | vflip | rotate90 |
+        ;; | 0 (000) - 0°       | false | false | false    | -> 2 (010) - V
+        ;; | 1 (001) - H        | true  | false | false    | -> 3 (011) - H + V
+        ;; | 2 (010) - V        | false | true  | false    | -> 0 (000) - R_0
+        ;; | 3 (011) - H+V      | true  | true  | false    | -> 1 (001) - H
+        ;; | 4 (100) - 90°      | false | false | true     | -> 5 (101) - H + R_90
+        ;; | 5 (101) - H+90°    | true  | false | true     | -> 4 (100) - R_90
+        ;; | 6 (110) - V+90°    | false | true  | true     | -> 7 (111) - H + V + R_90
+        ;; | 7 (111) - H+V+90°  | true  | true  | true     | -> 6 (110) - V + R_90
+        flipped-tiles (vec (for [y (range height)
+                                 x (range width)]
+                             (let [index (+ (* y width) x)
+                                   tile (nth tiles index)
+                                   hflip (:h-flip tile)
+                                   vflip (:v-flip tile)
+                                   rotate90 (:rotate90 tile)
+                                   current-state (bit-or (if hflip 1 0)
+                                                         (if vflip 2 0)
+                                                         (if rotate90 4 0))
+                                   new-state (case current-state
+                                               0 2          ; R_0 -> V
+                                               1 3          ; H -> H + V
+                                               2 0          ; V -> R_0
+                                               3 1          ; H + V -> H
+                                               4 5          ; R_90 -> H + R_90
+                                               5 4          ; H + R_90 -> R_90
+                                               6 7          ; V + R_90 -> H + V + R_90
+                                               7 6)         ; H + V + R_90 -> V + R_90
+                                   new-hflip (bit-test new-state 0)
+                                   new-vflip (bit-test new-state 1)
+                                   new-rotate90 (bit-test new-state 2)]
+                               {:x (:x tile)
+                                :y (:y tile)
+                                :tile (:tile tile)
+                                :h-flip new-hflip
+                                :v-flip new-vflip
+                                :rotate90 new-rotate90})))
+        reordered-tiles (vec (flatten (reverse (partition width flipped-tiles))))]
```

Comment:
in this case I use `partition` which can't be used with `rseq` 

>Caused by: java.lang.ClassCastException: class clojure.lang.LazySeq cannot be cast to class clojure.lang.Reversible (clojure.lang.LazySeq and clojure.lang.Reversible are in unnamed module of loader 'app')

Does it worth to wrap it into a vector like:
>(rseq (vec (partition width flipped-tiles)))
?

Reply:
Ah, sorry, I see what you mean. In this case it probably needs to remain as `reverse`, since `partition` does not return a `vector`. Sorry for the misleading feedback. You can't change it to `rseq` in this case, and calling `vec` on the `partition` result just so we can use `rseq` feels worse than `reverse`, at least to me.